src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1724,24 +1803,21 @@ uct_dc_mlx5_dci_keepalive_handle_failure(uct_dc_mlx5_iface_t *iface,
 
 ucs_status_t uct_dc_mlx5_iface_keepalive_init(uct_dc_mlx5_iface_t *iface)
 {
-    int full_handshake = iface->flags &
-                         UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE_FULL_HANDSHAKE;
-    ucs_status_t status;
-    uint8_t dci_index;
+    uint8_t sl, port_affinity, pool_index;
+    uct_dc_mlx5_dci_config_t dci_config;
 
     if (ucs_likely(iface->flags & UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE)) {
         return UCS_OK;
     }
 
-    dci_index = uct_dc_mlx5_iface_total_ndci(iface);
-    status    = uct_dc_mlx5_iface_create_dci(iface, 0, dci_index, 0,
-                                             full_handshake);
-    if (status != UCS_OK) {
-        return status;
-    }
+    sl             = iface->super.super.super.config.sl;
+    port_affinity  = iface->tx.port_affinity;
+
+    uct_dc_mlx5_init_dci_config_key(&dci_config, sl, port_affinity, 0, 1);
+    uct_dc_mlx5_dci_pool_get_or_create(iface, &dci_config, &pool_index);
 
     iface->flags        |= UCT_DC_MLX5_IFACE_FLAG_KEEPALIVE;
-    iface->keepalive_dci = dci_index;
+    iface->keepalive_dci = iface->tx.dci_counter - 1;
```

Comment:
Can this place be reached with `dci_counter` 0?

Reply:
No, in the current state of the PR a pool is being created with a single dci so dc_counter is at least 1.